datafusion/functions/src/unicode/substr.rs

```diff
@@ -168,10 +330,34 @@ where
     }
 }
 
+/// Extracts the substring of string starting at the start'th character, and extending for count characters if that is specified. (Same as substring(string from start for count).)
+/// substr('alphabet', 3) = 'phabet'
+/// substr('alphabet', 3, 2) = 'ph'
+/// The implementation uses UTF-8 code points as characters
+fn calculate_substr<'a, V, T>(string_array: V, args: &[ArrayRef]) -> Result<ArrayRef>
```

Comment:
I think this function adds an unecessary level of indirection 

Rather than having`substr` dispatch to `calculate_substr` which then dispatches to `calculate_string` or `calculate_string_view` I suspect the code would be easier to follow

Reply:
Thanks for pointing this out. I didn't realize it earlier.